### PR TITLE
feat - [webpack v5 migration]

### PR DESCRIPTION
**Pull request by christin-wednesday**
### Ticket Link
---------------------------------------------------

### Related Links
---------------------------------------------------
https://webpack.js.org/migrate/5/

### Description
---------------------------------------------------
1. Updated all plugins to the latest versions.
2. Tested, updated options and downgraded plugins to make everything work for Webpack 4

### Steps to Reproduce / Test
---------------------------------------------------
1. Try`yarn webpack configtest internals/webpack/webpack.config.prod.babel.js` to test the config.

### GIF's
---------------------------------------------------

### CODE REVIEW

File: internals/webpack/webpack.config.dev.babel.js [hunk 1]
```diff
@@ -20,7 +20,6 @@ module.exports = require('./webpack.base.babel')({
 
   // Don't use hashes in dev mode for better performance
   output: {
-    filename: '[name].js',
```
Comment (christin-wednesday) [inline internals/webpack/webpack.config.dev.babel.js:23]: This is the default value, the migration document suggested to remove it.


https://webpack.js.org/migrate/5/#clean-up-configuration

File: internals/webpack/webpack.config.prod.babel.js [hunk 2]
```diff
@@ -129,12 +124,6 @@ module.exports = require('./webpack.base.babel')({
           ios: true
         }
       ]
-    }),
-
-    new HashedModuleIdsPlugin({
-      hashFunction: 'sha256',
```
Comment (christin-wednesday) [inline internals/webpack/webpack.config.prod.babel.js:135]: https://webpack.js.org/migrate/5/#update-outdated-options

File: package.json [hunk 3]
```diff
@@ -193,20 +191,21 @@
...
+    "webpack": "^5.40.0",
+    "webpack-bundle-analyzer": "^4.4.2",
+    "webpack-cli": "4.7.2",
+    "webpack-dev-middleware": "5.0.0",
+    "webpack-hot-middleware": "2.25.0",
+    "webpack-pwa-manifest": "4.3.0",
     "whatwg-fetch": "3.0.0"
-  }
+  },
+  "sideEffects": false
```
Comment (christin-wednesday) [inline package.json:210]: https://webpack.js.org/guides/tree-shaking/#mark-the-file-as-side-effect-free

File: package.json [hunk 4]
```diff
@@ -68,13 +68,10 @@
     ]
   },
   "pre-commit": "lint:staged",
-  "resolutions": {
```
Comment (christin-wednesday) [inline package.json:71]: This feature requires an additional package called https://github.com/rogeriochaves/npm-force-resolutions

Comment (christin-wednesday) [reply package.json:71]: Removed the property since this was not working.

File: package.json [hunk 5]
```diff
@@ -113,40 +108,42 @@
...
-    "@babel/register": "7.4.0",
+    "@babel/plugin-transform-modules-commonjs": "7.14.5",
+    "@babel/plugin-transform-react-constant-elements": "7.14.5",
+    "@babel/plugin-transform-react-inline-elements": "7.14.5",
+    "@babel/polyfill": "^7.12.1",
+    "@babel/preset-env": "^7.14.7",
+    "@babel/preset-react": "^7.14.5",
+    "@babel/register": "7.14.5",
+    "@babel/runtime": "^7.14.6",
+    "@lcdp/offline-plugin": "^5.1.0",
```
Comment (christin-wednesday) [inline package.json:125]: https://github.com/NekR/offline-plugin/issues/497#issuecomment-832059275

**Review by alichherawalla — APPROVED** [review]

**Pull request merged**